### PR TITLE
Handle pointset messages

### PR DESCRIPTION
**Pull request by slevertbiot**
This is not a final review for pointset messages.  We still need to figure out what to do about:

- names vs DBO codes
- what fields are mandatory in the point sub document

### CODE REVIEW

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 1]
```diff
@@ -1,65 +1,146 @@
...
   }
 }
 
-function createSystemModelDocument(message: UdmiMessage): DeviceDocument {
-  const deviceDocument: DeviceDocument = createDefaultDeviceDocument(message);
+function createDeviceDocumentFromSystemModelDocument(
+  message: UdmiMessage,
+  builder: DeviceDocumentBuilder
+): DeviceDocument {
+  builder.section(message.data.location.section).site(message.data.location.site);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:39]: You went with the builder pattern, why return on another line? Just return line 39, but chain the `.build()`

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 2]
```diff
@@ -1,65 +1,146 @@
...
+  return builder.build();
 }
 
-function createSystemStateDocument(message: UdmiMessage): DeviceDocument {
-  const deviceDocument: DeviceDocument = createDefaultDeviceDocument(message);
+function createDeviceDocumentFromSystemStateDocument(
+  message: UdmiMessage,
+  builder: DeviceDocumentBuilder
+): DeviceDocument {
+  builder
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:48]: Same here, just return with `.build()`.

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 3]
```diff
@@ -1,65 +1,146 @@
...
+      .id(pointObject)
+      .name(pointObject)
+      .units(units)
+      .metaUnit(units)
+      .metaCode(pointObject)
+      .build();
+    points.push(point);
+  }
+
+  deviceBuilder.points(points);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:109]: Return with `.build()`.

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 4]
```diff
@@ -1,65 +1,146 @@
...
-  deviceDocument.make = message.data.hardware.make;
-  deviceDocument.model = message.data.hardware.model;
-  deviceDocument.operational = message.data.operational;
-  deviceDocument.serialNumber = message.data.serial_no;
-  deviceDocument.firmware = message.data.software.firmware;
+function createDeviceDocumentFromDefaultDeviceDocument(
+  message: UdmiMessage,
+  builder: DeviceDocumentBuilder
+): DeviceDocument {
+  builder.lastPayload(message.data.timestamp);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:63]: Return with `.build()`.

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 5]
```diff
@@ -1,65 +1,146 @@
...
+): DeviceDocument {
+  const points: Point[] = [];
+
+  for (var pointObject in message.data.points) {
+    const pointSetBuilder: PointBuilder = new PointBuilder();
+    const point: Point = pointSetBuilder.id(pointObject).name(pointObject).metaCode(pointObject).build();
+    points.push(point);
+  }
+
+  deviceBuilder.points(points);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:143]: Return with `.build()`.

File: udmif/event-handler/src/Point.ts [hunk 6]
```diff
@@ -0,0 +1,123 @@
+/**
+ * {
+            "id": "ZT-1",
```
Comment (mspratt-biot) [inline udmif/event-handler/src/Point.ts:3]: I'm assuming this too is an example to leave here? Maybe adjust the tabbing?

File: udmif/event-handler/src/DeviceDocumentFactory.ts [hunk 7]
```diff
@@ -137,10 +128,10 @@ function createDeviceDocumentFromPointsetConfig(
   for (var pointObject in message.data.points) {
     const pointSetBuilder: PointBuilder = new PointBuilder();
     const point: Point = pointSetBuilder.id(pointObject).name(pointObject).metaCode(pointObject).build();
+    console.log(point);
```
Comment (mspratt-biot) [inline udmif/event-handler/src/DeviceDocumentFactory.ts:131]: We keeping these console logs?

File: udmif/event-handler/src/Point.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+  value?: string;
+  units?: string;
+  meta?: {
+    code?: string;
+    units?: string;
+  };
+  state?: string;
+}
+
+export class PointBuilder {
```
[the remark this answers is not in the document]
Comment (slevertbiot) [reply udmif/event-handler/src/Point.ts:26]: For complex objects or having many fields, builder pattern is the way to go.  Very easy to use.

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**